crates/bevy_ui/src/widget/image.rs

```diff
@@ -8,6 +8,44 @@ use bevy_render::texture::Image;
 #[cfg(feature = "bevy_text")]
 use bevy_text::Text;
 
+#[derive(Clone)]
+pub struct ImageMeasure {
+    // target size of the image
+    size: Vec2,
+}
+
+impl Measure for ImageMeasure {
+    fn measure(
+        &self,
+        width: Option<f32>,
+        height: Option<f32>,
+        _: AvailableSpace,
+        _: AvailableSpace,
+    ) -> Vec2 {
+        let mut size = self.size;
+        match (width, height) {
+            (None, None) => {}
+            (Some(width), None) => {
+                size.y = width * size.y / size.x;
```

Comment:
I think a comment would be useful for this logic, I assume the goal is to maintain the aspect ratio of the image?